File: Source/FileHandling/read_nth_line.c
#include <stdio.h>
#include <stdlib.h>

// Function to read the n-th line of a file
char* read_nth_line(const char* file_path, int n) {
    FILE* file = fopen(file_path, "r");
    if(file == NULL) {
        printf("Could not open file %s\n", file_path);
        return NULL;
    }

    char* line = NULL;
    size_t len = 0;
    ssize_t read;

    int current_line = 0;
    while((read = getline(&line, &len, file)) != -1) {
        current_line++;
        if(current_line == n) {
            fclose(file);
            return line;
        }
    }

    // If we reach here, then the file has less than n lines
    fclose(file);
    free(line);
    return NULL;
}

File: Source/FileHandling/read_nth_line.js
const fs = require('fs');
const readline = require('readline');

function read_nth_line(file_path, n) {
    return new Promise((resolve, reject) => {
        const fileStream = fs.createReadStream(file_path);

        const rl = readline.createInterface({
            input: fileStream,
            crlfDelay: Infinity
        });

        let line_no = 0;

        rl.on('line', (line) => {
            line_no++;
            if(line_no === n) {
                resolve(line);
                rl.close();
            }
        });

        rl.on('close', () => {
            if(line_no < n) {
                reject(new Error(`The file has less than ${n} lines.`));
            }
        });
    });
}
